### PR TITLE
Issue #2: make pojo class that reperesents module metadata

### LINKED ISSUE #2: make pojo class that reperesents module metadata
just a pojo class need to be created.

### PR DESCRIPTION
**Pull request by gaurabdg**
Resolves #2
Simple POJO to represent module metadata

### CODE REVIEW

File: src/main/java/CheckProperty.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import java.util.Collections;
+import java.util.List;
+
+public class CheckProperty {
```
Comment (romani) [inline src/main/java/CheckProperty.java:4]: ModulePropertyDetails

File: src/main/java/Check.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import java.util.Collections;
+import java.util.List;
+
+public class Check {
```
Comment (romani) [inline src/main/java/Check.java:4]: ModuleDetails.

Everything in checkstyle config is module.

File: src/main/java/Check.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    private String name;
+    private String fullQualifiedName;
+    private String parent;
+    private String description;
+    private List<CheckProperty> propertyList;
```
Comment (romani) [inline src/main/java/Check.java:9]: `properties`

File: src/main/java/CheckProperty.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    private String name;
+    private String type;
+    private String defaultValue;
+    private String valueType;
+    private String description;
+    private List<String> propertyValues;
```
Comment (romani) [inline src/main/java/CheckProperty.java:10]: `values`

File: src/main/java/CheckProperty.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
+import java.util.Collections;
+import java.util.List;
+
+public class CheckProperty {
+    private String name;
+    private String type;
+    private String defaultValue;
+    private String valueType;
```
Comment (romani) [inline src/main/java/CheckProperty.java:8]: it is better to put ONLY for this field, javadoc to explain that it is for scalar/collection.

Comment (gaurabdg) [reply src/main/java/CheckProperty.java:8]: I didn't understand what you are trying to say, can you please state it again?

Comment (romani) [reply src/main/java/CheckProperty.java:8]: Please put javadoc over this field to explain reason of it.

**Pull request merged**